src/linesplitting.jl

```diff
@@ -19,53 +17,71 @@ coastlines(ga::GeoAxis)=split(coastlines(),ga)
 
 module LineSplitting
 
-	using GeometryBasics: LineString
-	using Makie: Observable, @lift
+	import GeometryBasics
+	import Makie: Observable, @lift
+	import GeoMakie: GeoAxis
+
     # Since we're overriding Base.split, we must import it
 	import Base.split
-	
-	function regroup(tmp::Vector)
-		coastlines_custom=LineString[]
-		println(typeof(coastlines_custom))
-		for ii in 1:length(tmp)
-			push!(coastlines_custom,tmp[ii][:]...)
-		end
-		coastlines_custom
+			
+	getlon(p::GeometryBasics.Point) = p[1]
+
+	###
+	function split(tmp::GeometryBasics.LineString, lon0::Real)
+		lon0<0.0 ? lon1=lon0+180 : lon1=lon0-180 
+
+		linenodes = GeometryBasics.coordinates(tmp)  # get coordinates of line nodes
+		# Find nodes that are on either side of lon0
+		cond = getlon.(linenodes) .>= lon1
+		# Find interval starts and ends
+		end_cond = diff(cond)  # nonzero values denote ends of intervals
+		end_inds = findall(!=(0), end_cond)
+		start_inds = [firstindex(linenodes);  end_inds .+ 1]  # starts of intervals
+		end_inds = [end_inds; lastindex(linenodes)]  # ends of intervals
+		# do the splitting
+		split_coords = view.(Ref(linenodes), UnitRange.(start_inds, end_inds))  # For each start-end pair, get those coords
+		# reconstruct lines from points
+		split_lines = GeometryBasics.LineString.(split_coords) 
 	end
-	
-	function split(tmp::Vector{<:LineString}, lon0::Real)
+
+	function split(tmp::Vector{<:GeometryBasics.LineString}, lon0::Real)
 		[split(a,lon0) for a in tmp]
 	end
-	
-	function split(tmp::LineString, lon0::Real)
-		lon0<0.0 ? lon1=lon0+180 : lon1=lon0-180 
-		np=length(tmp)
-		tmp2=fill(0,np)
-		for p in 1:np
-			tmp1=tmp[p]
-			tmp2[p]=maximum( [(tmp1[1][1]<=lon1)+2*(tmp1[2][1]>=lon1) , (tmp1[2][1]<=lon1)+2*(tmp1[1][1]>=lon1)] )
-		end
-		if !any(==(3), tmp2) # no value in tmp2 is equal to 3
-			[tmp]
-		else # some value in tmp2 is equal to 3
-			jj=[0;findall(tmp2.==3)...;np+1]
-			[LineString(tmp[jj[ii]+1:jj[ii+1]-1]) for ii in 1:length(jj)-1]
-		end
-	end
 
-	split(tmp::Vector,dest::Observable) = @lift(split(tmp, $(dest)))
-	split(tmp::Observable,dest::Observable) = @lift(split($(tmp), $(dest)))
-	split(tmp::Observable,dest::String) = @lift(split($(tmp), (dest)))
+	###
+	split(tmp::GeometryBasics.LineString,dest::Observable) = @lift(split(tmp, $(dest)))
+
+	function split(tmp::Vector{<:GeometryBasics.LineString},dest::Observable)
+		@lift([split(a,$(dest)) for a in tmp])
+	end
 
-	function split(tmp::Vector{<:LineString},dest::String)
+	###
+	split(tmp::GeometryBasics.LineString,ax::GeoAxis) = split(tmp, ax.dest)
+	
+	function split(tmp::Vector{<:GeometryBasics.LineString},ax::GeoAxis)
+		[split(a,ax.dest) for a in tmp]
```

Comment:
since ax.dest is an observable, isn't this sufficient?

```
split(tmp::GeometryBasics.LineString,dest::Observable) = @lift(split(tmp, $(dest)))
split(tmp::GeometryBasics.LineString,ax::GeoAxis) = split(tmp, ax.dest)
```

Reply:
An array of Observables probably won't be plottable by Makie - better to do the lifting as soon as possible IMO.

Reply:
not sure what you mean, but please go ahead